### PR TITLE
Fix for Schematics 3.2 dependencies

### LINKED ISSUE #11516: Unable to install spartacus libraries in a new angular app via schematics
**Describe the bug**
When I tried to add spartacus via schematics to a new shell app, I get the following:

>> ng add @spartacus/schematics
Your global Angular CLI version (10.2.1) is greater than your local
version (10.1.7). The local Angular CLI version is used.
To disable this warning use "ng config -g cli.warnings.versionMismatch false".
Installing packages for tooling via npm.
An unhandled exception occurred: npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! Found: @angular/compiler@10.1.6
npm ERR! node_modules/@angular/compiler
npm ERR!   peer @angular/compiler@"10.1.6" from @angular/compiler-cli@10.1.6
npm ERR!   node_modules/@angular/compiler-cli
npm ERR!     peer @angular/compiler-cli@"^10.0.0" from @angular-devkit/build-angular@0.1001.7
npm ERR!     node_modules/@angular-devkit/build-angular
npm ERR!       dev @angular-devkit/build-angular@"~0.1001.7" from the root project
npm ERR!     peer @angular/compiler-cli@"^10.0.0" from @ngtools/webpack@10.1.7
npm ERR!     node_modules/@ngtools/webpack
npm ERR!       @ngtools/webpack@"10.1.7" from @angular-devkit/build-angular@0.1001.7
npm ERR!       node_modules/@angular-devkit/build-angular
npm ERR!         dev @angular-devkit/build-angular@"~0.1001.7" from the root project
npm ERR!     1 more (the root project)
npm ERR!   peer @angular/compiler@"10.1.6" from @angular/platform-browser-dynamic@10.1.6
npm ERR!   node_modules/@angular/platform-browser-dynamic
npm ERR!     @angular/platform-browser-dynamic@"~10.1.6" from the root project
npm ERR!   3 more (codelyzer, the root project, @spartacus/schematics)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/compiler@"10.2.4" from @angular/platform-server@10.2.4
npm ERR! node_modules/@spartacus/schematics/node_modules/@angular/platform-server
npm ERR!   peer @angular/platform-server@"^10.1.0" from @nguniversal/express-engine@10.1.0
npm ERR!   node_modules/@spartacus/schematics/node_modules/@nguniversal/express-engine
npm ERR!     @nguniversal/express-engine@"^10.1.0" from @spartacus/schematics@3.1.1
npm ERR!     node_modules/@spartacus/schematics
npm ERR!       dev @spartacus/schematics@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!

I think the main error is here:

Could not resolve dependency:
npm ERR! peer @angular/common@“>=9.0.0 <10.0.0” from @ng-select/ng-select@4.0.4
npm ERR! node_modules/@ng-select/ng-select
npm ERR!   @ng-select/ng-select@“^4.0.0" from the root project

we are including ng-select 4 when our main development app uses 5.

**Tell us the version of Spartacus**
- Library version: 3.1
- Node 12
- npm 7.6.3
- ng 10.2.3

**To Reproduce**
1. Follow the steps from Spartacus docs to create a new angular app and add Spartacus via Schematics

**Expected behavior**
We should be able to install spartacus via schematics without errors

**Desktop (please complete the following information):**
 - OS: Mac OSX Big Sur

### PR DESCRIPTION
**Pull request by hackergil**
Fixes #11516

### CODE REVIEW

File: projects/schematics/package.json [hunk 1]
```diff
@@ -22,10 +22,10 @@
     "tslib": "^2.0.0"
   },
   "peerDependencies": {
-    "@angular-devkit/core": "^10.1.0",
```
Comment (marlass) [inline projects/schematics/package.json]: This looks like a breaking change. Is it needed?

Comment (hackergil) [reply projects/schematics/package.json]: Nope. I can revert it.

File: projects/schematics/package.json [hunk 2]
```diff
@@ -23,9 +23,9 @@
   },
   "peerDependencies": {
     "@angular-devkit/core": "^10.1.0",
-    "@angular-devkit/schematics": "^10.1.0",
-    "@angular/compiler": "^10.1.0",
-    "@schematics/angular": "^10.1.0",
+    "@angular-devkit/schematics": "^10.2.1",
+    "@angular/compiler": "^10.2.4",
+    "@schematics/angular": "^10.2.0",
```
Comment (marlass) [inline projects/schematics/package.json]: The comment was supposed to apply for all these deps ;)

**Review by marlass — CHANGES_REQUESTED** [review]

File: projects/schematics/package.json (hunk 2 above)
Comment (hackergil) [reply projects/schematics/package.json]: Done.

**Review by marlass — APPROVED** [review]

**Pull request merged**